ros_buildfarm/aggregate_status.py

```diff
@@ -0,0 +1,291 @@
+import os
+import re
+import yaml
+import requests
+import collections
+from .config import get_index as get_config_index, get_release_build_files
+
+"""
+
+  Aggregate Status takes the yaml generated by the different CPU architectures (e.g. ros_lunar_default, ros_lunar_ds)
+  and combines them into one yaml file per ROS distro. It also creates an English description of each package across
+  all of the architectures. See examples below.
+
+  Naming Conventions
+   * distro refers to ROS Distro (e.g. indigo)
+   * machine refers to different release builds (e.g. default, uxhf, dsv8)
+   * os_name refers to the name of the Operating System (e.g. ubuntu, debian)
+   * os_flavor refers to specific distros of the os (e.g. xenial, jessie)
+   * cpu refers to the cpu architecture (or source), (e.g. i386, amd64, source)
+   * candidate refers to different candidate builds, (e.g. build, test, main)
+   * combo refers to the os_flavor + the cpu
+
+  Example Descriptions:
+   - released - All candidates have the same version number
+   - waiting for new release - Build/test have the same version number, but there's nothing for main
+   - waiting for re-release - Build/test have the same version number, but there's an older version for main
+   - waiting for new/re-release - Build/test have the same version number, but there's either nothing or an older
+                                  version for main
+   - source builds, binary doesn't - If the source builds, but all the other cpus do not
+   - does not build on X - If the source and some of the cpus build, but some do not
```

Comment:
The descriptions seem to be biased to a single perspective. From an end user point of view "waiting for new/re-release" might make sense. But from a package maintainer point of view the terminology is misleading. E.g. in the case of "waiting for new release" the release has already been made. The only step missing is a sync of the testing repo to the main repo to make the new package available.

There I would recommend a different description more towards what is actually missing / needs to happen.

---

Just an idea: for the generated yaml it might be good to just use an "enum" value to describe the state and then however the data is being "viewed" can decide on the precise string.